tests/store/tracking/test_sqlalchemy_store.py

```diff
@@ -1470,9 +1470,11 @@ def _raise_exception_fn(*args, **kwargs):  # pylint: disable=unused-argument
 
         package = "mlflow.store.tracking.sqlalchemy_store.SqlAlchemyStore"
         with mock.patch(package + ".log_metric") as metric_mock, \
+                mock.patch(package + ".log_metrics") as metrics_mock, \
                 mock.patch(package + ".log_param") as param_mock, \
                 mock.patch(package + ".set_tag") as tags_mock:
             metric_mock.side_effect = _raise_exception_fn
+            metrics_mock.side_effect = _raise_exception_fn
```

Comment:
we should also add some additional tests to make sure that batched metric insertion is done correctly. For instance multiple metrics with the same key in a single batch, metrics with non-increasing steps, metrics from the batch **do not overwrite** existing metrics (with different keys or same keys if the existing ones have higher step values or timestamp) and **do overwrite** (existing metrics with same keys and lower timestamps and step values), ... etc.

Reply:
I added a couple of tests.